Source/init.cpp

```diff
@@ -47,7 +47,7 @@ void init_create_window()
 	if (!SpawnWindow(PROJECT_NAME, SCREEN_WIDTH, SCREEN_HEIGHT))
 		app_fatal("Unable to create main window");
 	dx_init(NULL);
-	atexit(dx_cleanup);
+	//atexit(dx_cleanup);

```

Comment:
This will defiantly affect other targets :)

Reply:
Right! I added it upon suggestion From Arczi84 i think because it seem to fix issues on the amiga. But actually the issue was in the freed semaphore being reused. Core review proves to be usefull once again because this changes should'nt be there. I'll correct my branch. Thank you for pointing this out.

BTW, are you dealing with the "AmigaDev" repo or the "Diasurgical" one above for this PR? (I'd tought Arczi juste send a PR to "AmigaDev" so that he can sync with my changes).

Reply:
My comments are in regards to merging thing in to Diasurgical, we have a goal of being fully BE compatible, so basic Amiga support would be a good way for us to achieve this. There is currently a PR for merging AmigaDev:master in to Diasurgical and as it it's good except for 3 bugs (save game support, crash and bad colors), so I have a bit of interest in the AmigaDev staying in a state where it can be merged in to Diasurgical :)

https://github.com/diasurgical/devilutionX/pull/397